jobsdb/jobsdb.go

```diff
@@ -1863,66 +1864,87 @@ func (jd *Handle) GetToProcess(ctx context.Context, params GetQueryParams, more
 
 var cacheParameterFilters = []string{"source_id", "destination_id"}
 
-func (jd *Handle) GetPileUpCounts(ctx context.Context) (map[string]map[string]int, error) {
+func (jd *Handle) GetPileUpCounts(ctx context.Context) error {
 	if !jd.dsMigrationLock.RTryLockWithCtx(ctx) {
-		return nil, fmt.Errorf("could not acquire a migration read lock: %w", ctx.Err())
+		return fmt.Errorf("could not acquire a migration read lock: %w", ctx.Err())
 	}
 	defer jd.dsMigrationLock.RUnlock()
 	if !jd.dsListLock.RTryLockWithCtx(ctx) {
-		return nil, fmt.Errorf("could not acquire a dslist read lock: %w", ctx.Err())
+		return fmt.Errorf("could not acquire a dslist read lock: %w", ctx.Err())
 	}
 	dsList := jd.getDSList()
 	jd.dsListLock.RUnlock()
-	statMap := make(map[string]map[string]int)
 
+	queryString := `with joined as (
+		select
+		  j.custom_val as customVal,
+		  j.workspace_id as workspace
+		from
+		  %[1]q j
+		  left join "v_last_%[2]s" s on j.job_id = s.job_id
+		where (
+		  s.job_state not in ('aborted', 'succeeded', 'migrated')
+		  or s.job_id is null
+		)
+	  )
+	  select
+		count(*),
+		customVal,
+		workspace
+	  from
+		joined
+	  group by
+		customVal,
+		workspace;`
+
+	g, ctx := errgroup.WithContext(ctx)
+	defaultConcurrency := 10
+	conc := jd.config.GetInt("jobsdb.pileupcount.parallelism", 10)
+	if conc <= 1 || conc > defaultConcurrency {
```

Comment:
A concurrency of 1 should be accepted. It might be useful in case we want to turn off concurrency completely.

Reply:
oh yes